File: gls_sync/validate.py
import re

SUPPORTED_COUNTRIES = frozenset({"BE", "NL", "LU"})

_EMAIL_RE = re.compile(r"^[^@\s]+@[^@\s]+\.[^@\s]+$")

_PHONE_PATTERNS = {
    "BE": re.compile(r"^(\+32|0032|0)\d{8,9}$"),
    "NL": re.compile(r"^(\+31|0031|0)\d{9}$"),
    "LU": re.compile(r"^(\+352|00352)\d{6,9}$"),
}


def validate_email(email: str) -> bool:
    if not email:
        return False
    return bool(_EMAIL_RE.match(email.strip()))


def _normalize_phone(phone: str) -> str:
    return re.sub(r"[\s\-().]", "", phone.strip())


def validate_phone(phone: str, country_code: str) -> bool:
    if not phone:
        return False
    normalized = _normalize_phone(phone)
    pattern = _PHONE_PATTERNS.get(country_code)
    if pattern is None:
        return False
    return bool(pattern.match(normalized))


def validate_country(country_code: str) -> bool:
    return country_code in SUPPORTED_COUNTRIES


def check_row(row: dict) -> list[str]:
    reasons = []
    if not validate_email(row.get("Email", "")):
        reasons.append("Email: ongeldig of ontbrekend")
    land = row.get("Land", "")
    if not validate_phone(row.get("Telefoon", ""), land):
        reasons.append("Telefoon: ongeldig formaat")
    if not validate_country(land):
        reasons.append("Land: niet ondersteund (verwacht BE, NL of LU)")
    return reasons

File: web/lib/validate.ts
export const SUPPORTED_COUNTRIES = new Set(["BE", "NL", "LU"]);

const EMAIL_RE = /^[^@\s]+@[^@\s]+\.[^@\s]+$/;

const PHONE_PATTERNS: Record<string, RegExp> = {
  BE: /^(\+32|0032|0)\d{8,9}$/,
  NL: /^(\+31|0031|0)\d{9}$/,
  LU: /^(\+352|00352)\d{6,9}$/,
};

export function validateEmail(email: string): boolean {
  if (!email) return false;
  return EMAIL_RE.test(email.trim());
}

function normalizePhone(phone: string): string {
  return phone.trim().replace(/[\s\-().]/g, "");
}

export function validatePhone(phone: string, countryCode: string): boolean {
  if (!phone) return false;
  const normalized = normalizePhone(phone);
  const pattern = PHONE_PATTERNS[countryCode];
  if (!pattern) return false;
  return pattern.test(normalized);
}

export function validateCountry(countryCode: string): boolean {
  return SUPPORTED_COUNTRIES.has(countryCode);
}

export interface OrderCheckInput {
  email: string;
  phone: string;
  countryCode: string;
}

export function checkOrder(order: OrderCheckInput): string[] {
  const reasons: string[] = [];
  if (!validateEmail(order.email)) {
    reasons.push("Email: ongeldig of ontbrekend");
  }
  if (!validatePhone(order.phone, order.countryCode)) {
    reasons.push("Telefoon: ongeldig formaat");
  }
  if (!validateCountry(order.countryCode)) {
    reasons.push("Land: niet ondersteund (verwacht BE, NL of LU)");
  }
  return reasons;
}
